src/base_home.h

```diff
@@ -0,0 +1,142 @@
+#pragma once
+#ifndef BASE_HOME
+#define BASE_HOME
+
+#include "submap.h"
+#include <unordered_map>
+#include "ammo.h"
+#include "string_id.h"
+
+struct submap;
+struct tripoint;
+class ammunition_type;
+
+using furn_id = int_id<furn_t>;
+using ammotype_id = string_id<ammunition_type>;
+
+/**
+ * Flags unique to a base.
+ */
+enum base_area_flag
+{
+    BASE_IN,   /**Inside of actual base*/
+    BASE_GND,  /**On base grounds. i.e. between perimeter fence and building*/
+    BASE_WALL, /**Terrain seperating BASE_IN and BASE_GND*/
+    BASE_EDGE, /**outer edge of base area*/
+    NO_NPC,    /**NPCs stay out*/
+};
+
+//furniture flags in json
+//BASE_CORE -> Is the very heart and soul of a base.
+//BASE_CMD# -> Level provided by control core.
+//CMD_SYS -> Acts as a data conduit for the command system.  If adjacent or forms a path to BASE_CORE, allows placement of aux systems.
+//AUX_SYS -> Is an auxiliary system. Must be built adjacent to BASE_CORE flag or CMD_SYS flag linked to a BASE_CORE flag.
+//Connectable -> Possible to install network module
+//NETWORKED -> Network enabled; able to form connection between command and remote system.
+
+/**
+ * Stores capacities of command system and the auxiliary systems present.
+ */
+struct command_sys
```

Comment:
Leave this whole structure for the very end. Don't define things like processing power until you have all the mundane stuff done.